src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -561,27 +571,35 @@ private bool DoesSolutionRequireAnInitialSaveAs()
             Debug.Assert(ThreadHelper.CheckAccess());
 
             // Check if user is doing File - New File without saving the solution.
-            var value = GetVSSolutionProperty((int)(__VSPROPID.VSPROPID_IsSolutionSaveAsRequired));
+            var value = GetVSSolutionProperty((int)(__VSPROPID.VSPROPID_IsSolutionSaveAsRequired)) ?? false;
             if ((bool)value)
             {
                 return true;
             }
 
             // Check if user unchecks the "Tools - Options - Project & Soltuions - Save new projects when created" option
-            value = GetVSSolutionProperty((int)(__VSPROPID2.VSPROPID_DeferredSaveSolution));
+            value = GetVSSolutionProperty((int)(__VSPROPID2.VSPROPID_DeferredSaveSolution)) ?? false;
             return (bool)value;
         }
 
         private object GetVSSolutionProperty(int propId)
         {
             Debug.Assert(ThreadHelper.CheckAccess());
+            object result = null;
+            var vsSolution = _serviceProvider.GetService<SVsSolution, IVsSolution>();
 
-            object value;
-            int hr = _vsSolution.GetProperty(propId, out value);
+            if (vsSolution != null)
+            {
+                int hr = vsSolution.GetProperty(propId, out result);
 
-            ErrorHandler.ThrowOnFailure(hr);
+                ErrorHandler.ThrowOnFailure(hr);
+            }
+            else
+            {
+                Debug.Fail("Failed to get IVsSolution from service provider.");
```

Comment:
Not needed here this alone may generate tens of meaningless annoying asserts. Generally speaking using Debug.Assert is a bad practice. We shouldn't avoid it whenever is possible.